src/Stratis.SmartContracts.CLR/Loader/ContractAssembly.cs

```diff
@@ -49,12 +49,16 @@ private Type GetDeployedType()
                     x.CustomAttributes.Any(y => y.AttributeType == typeof(DeployAttribute)));
         }
 
-        public static bool IsContractType(Type typeDefinition)
+        private static bool IsContractType(Type typeDefinition)
         {
             return typeDefinition.IsClass &&
                    !typeDefinition.IsAbstract &&
-                   typeDefinition.BaseType != null &&
-                   typeDefinition.BaseType == typeof(SmartContract);
+                   InheritsFromType(typeDefinition, typeof(SmartContract));
```

Comment:
Thanks for letting me know 👍 
Updated

Reply:
@drmathias Is this code is going to be used only by validator ? Can you check this ? Is there any other place it is used in fullnode ? We want to ensure that this code change will not cause consensus issues on network.

Reply:
This code is only referenced to set the `ContractAssembly.DeployedType`, which is used by the contract swagger doc generator and `ReflectionVirtualMachine.Create`. I assume this would cause consensus issues, if a smart contract with 2 levels of inheritance was created on the network. In the example referenced above, updated versions of the node would execute the constructor of `NonFungibleTicket` while older versions of the node would execute the constructor of `NonFungibleToken`.

Reply:
I think this assumption is not correct. Tyler is able to deploy and run inherited types indirectly already.

Reply:
This is because he's creating a deployer contract, which directly inherits from `SmartContract`. To clarify, I mean that it could cause consensus issues, if a smart contract with 2 levels of inheritance, which was decorated with the `[Deploy]` attribute was created on the network.